### ISSUE TITLE
Cors plugin breaks content-type mapping.

### ISSUE THREAD

**Issue by ApproachingApathy** [elysiajs/elysia#40]
Using the cors plugin causes Elysia not to properly set content-type to "application/json" when an object is returned from a handler. After looking into it a bit it seems like Elysia is currently not setting the content-type if *any* headers are present. It might make more sense if content-type is always set unless a content-type header is present. I believe this is controlled by `handlers.ts mapEarlyResponse` and `handlers.ts mapResponse`. I'd be interested in working on this if desired.

**Comment by SaltyAom** [comment]
Hi, thanks for reporting the problem in.

I need a bit more information to replicate this problem on my end.

Can you provide me with your version of the following:
- Elysia
- `@elysiajs/cors`

A replicable code snippet that causes this problem.

Thanks a lot!

**Comment by ApproachingApathy** [comment]
```ts
import { Elysia } from "elysia";
import { swagger } from "@elysiajs/swagger";
import { cors } from "@elysiajs/cors"

const app = new Elysia()
	.use(
		swagger({
			path: "/docs",
		})
	)
	.use(cors())
	.get("/", () => ({ "same": "a" }))


app.listen(3000);

console.log(`Elysia is running at ${app.server?.hostname}:${app.server?.port}`);

export type App = typeof app;
```
@elysiajs/cors: "0.3.0"
elysia: "0.4.0"

If I comment out `.use(cors)` content-type is `application/json` if I leave it in `text/plain`

**Comment by SaltyAom** [comment]
This should be fixed on Elysia 0.5.1 with `@elysiajs/cors` 0.5.0, can update and try it out again to see if it's fixed.

**Comment by ApproachingApathy** [comment]
Thanks. This appears to be fixed!
